test/jdk/com/sun/jndi/ldap/LdapPoolTimeoutTest.java

```diff
@@ -109,38 +107,55 @@ public void test() throws Exception {
-            if (msg != null &&
-                    (msg.contains("Network is unreachable")
-                            || msg.contains("No route to host")
-                            || msg.contains("Connection timed out"))) {
-                // got the expected exception
-                System.out.println("Received expected RuntimeException message: " + msg);
-            } else {
-                // propagate the unexpected exception
-                throw e;
+            final InitialDirContext unexpectedCtx =
+                    LdapTimeoutTest.assertCompletion(CONNECT_MILLIS - 1000,
+                            2 * CONNECT_MILLIS + TOLERANCE,
+                            () -> new InitialDirContext(env));
+            throw new RuntimeException("InitialDirContext construction was expected to fail," +
+                    " but returned " + unexpectedCtx);
+        } catch (Throwable t) {
+            final NamingException namingEx = findNamingException(t);
+            if (namingEx != null) {
+                // found the NamingException, verify it's the right reason
+                if (namingEx.getCause() instanceof SocketTimeoutException ste) {
+                    // got the expected exception
+                    System.out.println("Received expected SocketTimeoutException: " + ste);
+                    return;
+                }
+                // rely on the exception message to verify the expected exception
+                final String msg = namingEx.getCause() == null
+                        ? namingEx.getMessage()
+                        : namingEx.getCause().getMessage();
+                if (msg != null &&
+                        (msg.contains("Network is unreachable")
+                                || msg.contains("No route to host")
+                                || msg.contains("Timed out waiting for lock")
+                                || msg.contains("Connect timed out")
+                                || msg.contains("Timeout exceeded while waiting for a connection"))) {
+                    // got the expected exception
+                    System.out.println("Received expected NamingException with message: " + msg);
+                    return;
+                }
             }
-        } catch (NamingException ex) {
-            final String msg = ex.getCause() == null ? ex.getMessage() : ex.getCause().getMessage();
-            if (msg != null &&
-                    (msg.contains("Network is unreachable")
-                        || msg.contains("Timed out waiting for lock")
-                        || msg.contains("Connect timed out")
-                        || msg.contains("Timeout exceeded while waiting for a connection"))) {
-                // got the expected exception
-                System.out.println("Received expected NamingException message: " + msg);
+            // unexpected exception, propagate it
+            if (t instanceof Exception e) {
+                throw e;
```

Comment:
I suggest to see if replacing `catch (Throwable t) {` with `catch (Exception | Error t) {` would work - because it might allow you to simply rethrow with `throw t` without having to rewrap.
I am not %100 sure the compiler will be smart enough though :-)